engine/lib/ast_builder.ml

```diff
@@ -22,17 +22,173 @@ module Make (F : Features.T) = struct
     let ty_tuple_or_id : ty list -> ty = function
       | [ ty ] -> ty
       | types -> ty_tuple types
+
+    let ty_control_flow ~(continue_type : ty) ~(break_type : ty) : ty =
+      TApp
+        {
+          ident = Global_ident.of_name Type Core__ops__control_flow__ControlFlow;
+          args = [ GType break_type; GType continue_type ];
+        }
+
+    let ty_cf_return ~(acc_type : ty) ~(break_type : ty)
```

Comment:
We need a comment that explains we encode `return` as `Break(Break(...))` and `break` as `Break(Continue)`, to justify the existence of this funciton